### PR TITLE
#175 view company sensitive data

### PR DESCRIPTION
**Pull request by Shushunya**
## Summary of issue

Issue #175 
An authorized user is able to view the page of any Company with hidden contact data of the company.
Once the user presses the buttons "Show phone" or "Show email", the contact data should be shown and the data of the user who provided this action should be logged into the log journal.

## Summary of change

Two Profile serializers added: 
- `ProfileSensitiveDataROSerializer`: ReadOnly serializer, returns phone number and email
- `ProfileDetailSerializer`: ReadOnly serializer, returns detail data on profile without email and phone number

`ProfileDetailAPIView` altered:
- `get_serializer_class` method overriden
- all serializer calls in the methods adjusted
- `retrieve` method altered

upd: 
- `get_queryset` removed
- `get_serializer_class` returns proper serializer for the owner on GET request now
- `permission_classes` altered
- custom permission `UserIsProfileOwnerOrReadOnly` added
- `destroy` -> `perform_destroy`
- overriden `retrieve` and `update` removed

Three tests added and four are adjusted.

tests cleaned up a bit:
- setUpTestData moved to setUp in `test_crud_profiles`
- redundant code removed
- urls adjusted

### CODE REVIEW

File: profiles/views.py [hunk 1]
```diff
@@ -96,33 +95,47 @@ def get_queryset(self, pk=None):
         if self.request.method in ['PUT', 'PATCH']:
             return Profile.objects.filter(profile_id=pk)
 
+    def get_serializer_class(self):
+        get_contacts = self.request.query_params.get("get_contacts")
```
Comment (ProDG) [inline profiles/views.py:99]: I suggest to rename it a bit. `get_contacts` → `with_contacts`.

File: profiles/views.py [hunk 2]
```diff
@@ -96,33 +95,47 @@ def get_queryset(self, pk=None):
...
+        instance = self.get_object()
+        if not request.user.is_authenticated and request.query_params.get("get_contacts"):
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+
+        if request.user.id == instance.person.id:
+            serializer = ProfileSerializer(instance)
+        else:
+            serializer = self.get_serializer(instance)
+
+        return Response(serializer.data)
```
Comment (ProDG) [inline profiles/views.py:115]: Sorry, but this is not DRF code.

File: profiles/views.py [hunk 3]
```diff
@@ -96,33 +95,47 @@ def get_queryset(self, pk=None):
...
+        get_contacts = self.request.query_params.get("get_contacts")
+        if self.request.method == 'GET':
+            return ProfileSensitiveDataROSerializer if get_contacts else ProfileDetailSerializer
+        else:
+            return ProfileSerializer
+
+    def retrieve(self, request, *args, **kwargs):
+        instance = self.get_object()
+        if not request.user.is_authenticated and request.query_params.get("get_contacts"):
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
```
Comment (ProDG) [inline profiles/views.py:108]: Permissions must be checked in permission classes.

Comment (Shushunya) [reply profiles/views.py:108]: That's a temporary solution until we have permission for the owner created.

File: profiles/views.py (hunk 1 above)
Comment (Shushunya) [reply profiles/views.py]: Done. Thanks :)

File: profiles/views.py (hunk 2 above)
Comment (Shushunya) [reply profiles/views.py]: Fixed.

File: profiles/views.py (hunk 3 above)
Comment (Shushunya) [reply profiles/views.py]: Permission is added and this is fixed now.

**Pull request merged**